src/sagemaker/analytics.py

```diff
@@ -740,5 +742,6 @@ def _search(self, search_expression, sort_by, sort_order):
                 search_args["NextToken"] = search_response["NextToken"]
             else:
                 break
+            time.sleep(self.POLL)
```

Comment:
1. How long it may takes to sleep here? 10 seconds * by what? 
2. If we want to prevent throttling, can we implement an exponnential backoff for the API call?

Reply:
Yeah, 10 secs should be too much wait, second thinking 0.5 sec should be sufficient assist with 4 retries. It is one customer face throttling Exception and hard to reproduce a throttling call